code/modules/projectiles/projectile/beams.dm

```diff
@@ -25,6 +45,13 @@
 	icon_state = "heavylaser"
 	damage = 40
 
+/obj/item/projectile/beam/heavylaser/on_hit(atom/target, blocked = 0)

```

Comment:
You should consider making the /heavylaser a child of /laser to avoid copypase. Also consider the firestacks being proportional to the damage.